### DISCUSSION TITLE
Pressing "Enter" should be disabled once onSubmit is called

### DISCUSSION THREAD

**Discussion by m00gl3** [ritikbanger/react18-input-otp#26] [Q&A] [answered]
Hello again,
I set numInputs to be 4

Once I enter the 4 digits, it runs onSubmit.
However, If I quickly press Enter, while it is running, it tries to run onSubmit again but with an empty value which is erroneous to the process. (I'm using a loader/overlay and am still able to quickly press "Enter")

Solution: Once you enter the number of digits and automatically call onSubmit. Disable pressing "Enter" key

Thank you!

**Comment by ritikbanger** [comment]
Hello @m00gl3, you can use the onKeyDown prop to modify the behavior. You can look for event of enter key and prevent the submission in your code itself.

Or please share your code so that I can help you with the direction. 

This does not seem to be a bug and looks more of a help you need to achieve what you are doing. I am moving this to a discussion. Please share a minimal reproducable codesandbox or a video or share your code, so that I can help you with the right direction. Thanks.

**Comment by m00gl3** [comment, marked as answer]
You were right, this was a bug on my end. I implemented something incorrectly on my end
